// react/esm-production.js
import {
  require_react
} from "../chunk-3D5M5TPY.js";
const _esmModule = require_react();
export const Children = _esmModule.Children;
export const Component = _esmModule.Component;
export const Fragment = _esmModule.Fragment;
export const Profiler = _esmModule.Profiler;
export const PureComponent = _esmModule.PureComponent;
export const StrictMode = _esmModule.StrictMode;
export const Suspense = _esmModule.Suspense;
export const SuspenseList = _esmModule.SuspenseList;
export const cloneElement = _esmModule.cloneElement;
export const createContext = _esmModule.createContext;
export const createElement = _esmModule.createElement;
export const createFactory = _esmModule.createFactory;
export const createRef = _esmModule.createRef;
export const forwardRef = _esmModule.forwardRef;
export const isValidElement = _esmModule.isValidElement;
export const lazy = _esmModule.lazy;
export const memo = _esmModule.memo;
export const startTransition = _esmModule.startTransition;
export const useCallback = _esmModule.useCallback;
export const useContext = _esmModule.useContext;
export const useDebugValue = _esmModule.useDebugValue;
export const useDeferredValue = _esmModule.useDeferredValue;
export const useEffect = _esmModule.useEffect;
export const useId = _esmModule.useId;
export const useImperativeHandle = _esmModule.useImperativeHandle;
export const useInsertionEffect = _esmModule.useInsertionEffect;
export const useLayoutEffect = _esmModule.useLayoutEffect;
export const useMemo = _esmModule.useMemo;
export const useReducer = _esmModule.useReducer;
export const useRef = _esmModule.useRef;
export const useState = _esmModule.useState;
export const useSyncExternalStore = _esmModule.useSyncExternalStore;
export const useTransition = _esmModule.useTransition;
export const version = _esmModule.version;
export default _esmModule;//# sourceMappingURL=http://go/sourcemap/sourcemaps/dc8361355d709f306d5159635a677a571b277bc0/node_modules/react/esm-production.js.map
